Fixes leave organisation button error

SriramPatibanda opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**

BugFix

**Did you add tests for your changes?**

No

**If relevant, did you update the documentation?**

No

**Summary**

Updates the app such that when a user isn't a part of any organization, a snackbar with a warning message is displayed. Adds joinedOrgs variable and updates FlatButton to TextButton at places as FlatButton is now deprecated.

Fixes #176 

**Does this PR introduce a breaking change?**

No

**Other information**

N.A.

SriramPatibanda commented:
@Sagar2366 Please review this PR whenever possible

Sagar2366 commented:
If user is not part of any organization then remove leave organization option for the user or disable it.

SriramPatibanda commented:
> If user is not part of any organization then remove leave organization option for the user or disable it.

Yeah, I was thinking of proposing but did this anyway. Will make the necessary changes soon.

SriramPatibanda commented:
@Sagar2366 The requested changes have been made, kindly review them whenever possible. I also removed the variable that I added in my previous commit i.e. joinedOrgs as I understood that it was unnecessary.

SriramPatibanda commented:
@Sagar2366 along with this, I think even the switch organization button can be made such that it's shown only when the user in more than 1 organisations. What do you think? Should I cover even that in this PR or do you want me to make a separate PR if you think that this change is necessary?

Sagar2366 commented:
@SriramPatibanda pull latest changes from master, this feature is already added.
In recent changes this functionality is also added - if user is part of multiple organization and leaves current organization then next organization from joined group will be automatically be considered as current organization. But as you are on older code, it is showing null, and also not removing leave organization button. Please test this scenario again on your PR after pulling master changes.

SriramPatibanda commented:
@Sagar2366 I've pulled the latest changes and there are some errors in the code. If you observe, the register_form.dart file has commented code at the bottom of the file. The lastest PR that has been merged seems to have some issues.

SriramPatibanda commented:
@Sagar2366 I've fixed the error and can push the changes to this repo if you permit me to.

SriramPatibanda commented:
> @SriramPatibanda pull latest changes from master, this feature is already added.
> In recent changes this functionality is also added - if user is part of multiple organization and leaves current organization then next organization from joined group will be automatically be considered as current organization. But as you are on older code, it is showing null, and also not removing leave organization button. Please test this scenario again on your PR after pulling master changes.

Yeah the changes you mentioned are indeed there, but what I suggested was something a little different. Whenever a user is there in only one organisation, there's technically no need to show the switch organization button as he's there in only 1 org. So I suggest that show the switch org button only when the user in 2 or more orgs.  Let me know what you think about it.

SriramPatibanda commented:
I've fixed the errors in the register form and the code is ready to be reviewed and merged.

Sagar2366 commented on lib/views/pages/login_signup/register_form.dart:
```diff
@@ -471,7 +465,8 @@ class RegisterFormState extends State<RegisterForm> {
         });
   }
-  /*_successToast(String msg) {
+
```
This changes are already added in PR https://github.com/PalisadoesFoundation/talawa/pull/373

SriramPatibanda replied:
When I made the PR, they weren't and that's why I did them. No problem, I'll just revert these changes, pull and push.

SriramPatibanda replied:
@Sagar2366 , I've made the required changes. Kindly review them.
